app/src/main/java/de/westnordost/streetcomplete/quests/traffic_signals_vibrate/AddTrafficSignalsVibration.kt

```diff
@@ -1,28 +1,48 @@
 package de.westnordost.streetcomplete.quests.traffic_signals_vibrate
 
+import de.westnordost.osmapi.map.MapDataWithGeometry
+import de.westnordost.osmapi.map.data.Element
 import de.westnordost.streetcomplete.R
+import de.westnordost.streetcomplete.data.elementfilter.toElementFilterExpression
 import de.westnordost.streetcomplete.data.meta.updateWithCheckDate
-import de.westnordost.streetcomplete.data.osm.osmquest.OsmFilterQuestType
 import de.westnordost.streetcomplete.data.osm.changes.StringMapChangesBuilder
+import de.westnordost.streetcomplete.data.osm.osmquest.OsmElementQuestType
 import de.westnordost.streetcomplete.ktx.toYesNo
 
-class AddTrafficSignalsVibration : OsmFilterQuestType<Boolean>() {
+class AddTrafficSignalsVibration : OsmElementQuestType<Boolean> {
 
-    override val elementFilter = """
+    private val crossingFilter by lazy { """
         nodes with crossing = traffic_signals and highway ~ crossing|traffic_signals and foot!=no
         and (
           !$VIBRATING_BUTTON
           or $VIBRATING_BUTTON = no and $VIBRATING_BUTTON older today -4 years
           or $VIBRATING_BUTTON older today -8 years
         )
-    """
+    """.toElementFilterExpression() }
+
+    private val excludedWaysFilter by lazy { """
+        ways with
+          highway = cycleway and foot !~ yes|designated
```

Comment:
There are many other combinations with foot - permissive etc.
Does it make sense to include those? 🤷

Reply:
Hmm... I guess a crossing with foot=permissive cycleway is unlikely to have sound signals as it's mainly for cyclists
plus, this way it's constistent with the tactile paving quest: https://github.com/streetcomplete/StreetComplete/blob/master/app/src/main/java/de/westnordost/streetcomplete/quests/tactile_paving/AddTactilePavingCrosswalk.kt#L30

Reply:
ok